src/sys/unix/selector/poll.rs

```diff
@@ -0,0 +1,725 @@
+
+        let res = syscall!(poll(
+            fds.as_mut_ptr() as *mut libc::pollfd,
+            fds.len() as libc::nfds_t,
+            timeout_ms,
+        ));
+
+        match res {
+            Ok(num_events) => break Ok(num_events as usize),
+            // poll returns EAGAIN if we can retry it.
+            Err(e) if e.raw_os_error() == Some(libc::EAGAIN) => continue,
+            Err(e) => return Err(e),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Event {
+    token: Token,
+    events: libc::c_short,
+}
+
+pub type Events = Vec<Event>;
+
+pub mod event {
+    use crate::sys::unix::selector::poll::POLLRDHUP;
+    use crate::sys::Event;
+    use crate::Token;
+    use std::fmt;
+
+    pub fn token(event: &Event) -> Token {
+        event.token
+    }
+
+    pub fn is_readable(event: &Event) -> bool {
+        (event.events & libc::POLLIN) != 0 || (event.events & libc::POLLPRI) != 0
+    }
+
+    pub fn is_writable(event: &Event) -> bool {
+        (event.events & libc::POLLOUT) != 0
+    }
+
+    pub fn is_error(event: &Event) -> bool {
+        (event.events & libc::POLLERR) != 0
+    }
+
+    pub fn is_read_closed(event: &Event) -> bool {
+        // Both halves of the socket have closed
+        (event.events & libc::POLLHUP) != 0
+            // Socket has received FIN or called shutdown(SHUT_RD)
+            || (event.events & POLLRDHUP) != 0
+    }
+
+    pub fn is_write_closed(event: &Event) -> bool {
+        // Both halves of the socket have closed
+        (event.events & libc::POLLHUP) != 0
+            // Unix pipe write end has closed
+            || ((event.events & libc::POLLOUT) != 0 && (event.events & libc::POLLERR) != 0)
+            // The other side (read end) of a Unix pipe has closed.
+            || (event.events == libc::POLLERR)
+    }
+
+    pub fn is_priority(event: &Event) -> bool {
+        (event.events & libc::POLLPRI) != 0
+    }
+
+    pub fn is_aio(_: &Event) -> bool {
+        // Not supported in the kernel, only in libc.
+        false
+    }
+
+    pub fn is_lio(_: &Event) -> bool {
+        // Not supported.
+        false
+    }
+
+    pub fn debug_details(f: &mut fmt::Formatter<'_>, event: &Event) -> fmt::Result {
+        #[allow(clippy::trivially_copy_pass_by_ref)]
+        fn check_events(got: &libc::c_short, want: &libc::c_short) -> bool {
+            (*got & want) != 0
+        }
+        debug_detail!(
+            EventsDetails(libc::c_short),
+            check_events,
+            libc::POLLIN,
+            libc::POLLPRI,
+            libc::POLLOUT,
+            libc::POLLRDNORM,
+            libc::POLLRDBAND,
+            libc::POLLWRNORM,
+            libc::POLLWRBAND,
+            libc::POLLERR,
+            libc::POLLHUP,
+        );
+
+        // Can't reference fields in packed structures.
+        let e_u64 = event.token.0;
+        f.debug_struct("epoll_event")
+            .field("events", &EventsDetails(event.events))
+            .field("u64", &e_u64)
```

Comment:
Little copy and paste fix ;)
```suggestion
        f.debug_struct("poll_event")
            .field("token", &event.token)
            .field("events", &EventsDetails(event.events))
```